timelock-corruption-detection/src/main/java/com/palantir/timelock/corruption/detection/CorruptionDetector.java

```diff
@@ -17,5 +17,5 @@
 package com.palantir.timelock.corruption.detection;
 
 public interface CorruptionDetector {
-    boolean hasDetectedCorruption();
+    boolean shootTimeLock();
```

Comment:
maybe `shouldRejectRequests()`? The impression I get if I come across a method like this is that it actually makes the server stop requests and returns true if successful/false if not